### PR TITLE
Fix Diagnostics.Tracing ILLinkTrim entries

### PR DESCRIPTION
**Pull request by eerhardt**
- PropertyValue/ReferenceTypeHelper's pattern is recognized by the ILLinker. Protected it with a new trimming test.
- RuntimeEventSource doesn't need to be rooted. This class is used by IL and doesn't get trimmed.
- EventPipe* and NativeRuntimeEventSource only need to be rooted in CoreCLR, since they aren't used in Mono. So they are moved to just CoreCLR's ILLinkTrim file.

Contributes to #35199

cc @marek-safar

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @noahfalk

File: src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml [hunk 1]
```diff
@@ -37,15 +37,6 @@
       <!-- Methods is used by VS Tasks Window. -->
       <method name="GetActiveTaskFromId" />
     </type>
-    <!-- Accessed via private reflection by tracing controller. -->
-    <type fullname="System.Diagnostics.Tracing.EventPipe*" />
-    <!-- Accessed via private reflection and by native code. -->
-    <type fullname="System.Diagnostics.Tracing.RuntimeEventSource" />
```
Comment (joperezr) [inline src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml:43]: What about this one, it didn't seem to get moved or gotten a new anotation

Comment (joperezr) [reply src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml:43]: oh nvm, I just saw your description on the PR, sorry my bad.

**Review by joperezr — APPROVED** [review]

**Review by marek-safar — APPROVED** [review]

**Review by noahfalk — APPROVED** [review]
👍

**Comment by noahfalk** [comment]
cc @dotnet/dotnet-diag

**Review by sywhang — APPROVED** [review]

**Pull request merged**